std/traits.d

```diff
@@ -5180,10 +5180,10 @@ enum isAssignable(Lhs, Rhs = Lhs) = isRvalueAssignable!(Lhs, Rhs) && isLvalueAss
 }
 
 // ditto
-private enum isRvalueAssignable(Lhs, Rhs = Lhs) = __traits(compiles, lvalueOf!Lhs = rvalueOf!Rhs);
+private enum isRvalueAssignable(Lhs, Rhs = Lhs) = __traits(compiles, { lvalueOf!Lhs = rvalueOf!Rhs; });
```

Comment:
Isn't this a more of a workaround for a compiler issue?

Reply:
I'm conflicted about this, because I'm not exactly sure of what code path is taken within the compiler.
However, I can see why this behavior (CTFE-ing the expression) would make sense. But it's also a fatal flaw with `__traits(compiles, ...)`, which is not shared by `is(typeof(...))`. On a related not, I never understood why we added the former if we had the latter.